internal/pkg/config/input.go

```diff
@@ -0,0 +1,96 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package config
+
+import (
+	"fmt"
+	"time"
+)
+
+// ServerTimeouts is the configuration for the server timeouts
+type ServerTimeouts struct {
+	Read  time.Duration `config:"read"`
+	Write time.Duration `config:"write"`
+}
+
+// InitDefaults initializes the defaults for the configuration.
+func (c *ServerTimeouts) InitDefaults() {
+	c.Read = 5 * time.Second
+	c.Write = 60 * 10 * time.Second // 10 minutes (long poll)
+}
+
+// ServerProfile is the configuration for profiling the server.
+type ServerProfile struct {
+	Bind string `config:"bind"`
+}
+
+// InitDefaults initializes the defaults for the configuration.
+func (c *ServerProfile) InitDefaults() {
+	c.Bind = "localhost:6060"
+}
+
+// ServerTLS is the TLS configuration for running the TLS endpoint.
+type ServerTLS struct {
+	Key  string `config:"key"`
+	Cert string `config:"cert"`
+}
+
+// Server is the configuration for the server
+type Server struct {
+	Host              string         `config:"host"`
+	Port              uint16         `config:"port"`
+	TLS               ServerTLS      `config:"tls"`
+	Timeouts          ServerTimeouts `config:"timeouts"`
+	MaxHeaderByteSize int            `config:"max_header_byte_size"`
+	RateLimitBurst    int            `config:"rate_limit_burst"`
+	RateLimitInterval time.Duration  `config:"rate_limit_interval"`
+	MaxEnrollPending  int64          `config:"max_enroll_pending"`
+	Profile           ServerProfile  `config:"profile"`
+}
+
+// InitDefaults initializes the defaults for the configuration.
+func (c *Server) InitDefaults() {
+	c.Host = "0.0.0.0"
+	c.Port = 8000
```

Comment:
different defaults?  above is: localhost:6000

Reply:
The one above is for the profiler, not for the HTTP endpoint.